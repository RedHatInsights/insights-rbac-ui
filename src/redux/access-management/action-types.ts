export const GET_PRINCIPAL_ACCESS = 'GET_PRINCIPAL_ACCESS' as const;
