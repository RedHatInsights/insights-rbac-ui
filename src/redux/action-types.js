export const FETCH_GROUP = 'FETCH_GROUP';
export const FETCH_SYSTEM_GROUP = 'FETCH_SYSTEM_GROUP';
export const FETCH_GROUPS = 'FETCH_GROUPS';

export const ADD_GROUP = 'ADD_GROUP';
export const UPDATE_GROUP = 'UPDATE_GROUP';
export const REMOVE_GROUPS = 'REMOVE_GROUPS';

export const FETCH_USERS = 'FETCH_USERS';

export const ADD_ROLE = 'ADD_ROLE';
export const FETCH_ROLE = 'FETCH_ROLE';
export const FETCH_ROLES = 'FETCH_ROLES';
export const REMOVE_ROLE = 'REMOVE_ROLE';
export const RESET_SELECTED_ROLE = 'RESET_SELECTED_ROLE';
export const FETCH_ROLES_FOR_WIZARD = 'FETCH_ROLES_FOR_WIZARD';

export const FETCH_GROUP_POLICIES = 'FETCH_GROUP_POLICIES';
export const FETCH_POLICY = 'FETCH_POLICY';
export const ADD_POLICY = 'ADD_POLICY';
export const UPDATE_POLICY = 'UPDATE_POLICY';
export const REMOVE_POLICY = 'REMOVE_POLICY';
export const RESET_SELECTED_GROUP = 'RESET_SELECTED_GROUP';

export const REMOVE_MEMBERS_FROM_GROUP = 'REMOVE_MEMBERS_FROM_GROUP';
export const ADD_MEMBERS_TO_GROUP = 'ADD_MEMBERS_TO_GROUP';

export const REMOVE_ROLES_FROM_GROUP = 'REMOVE_ROLES_FROM_GROUP';
export const ADD_ROLES_TO_GROUP = 'ADD_ROLES_TO_GROUP';
export const FETCH_ROLES_FOR_GROUP = 'FETCH_ROLES_FOR_GROUP';
export const FETCH_MEMBERS_FOR_GROUP = 'FETCH_MEMBERS_FOR_GROUP';
export const FETCH_ADD_ROLES_FOR_GROUP = 'FETCH_ADD_ROLES_FOR_GROUP';
export const FETCH_ROLE_FOR_USER = 'FETCH_ROLE_FOR_USER';
export const FETCH_ROLE_FOR_PRINCIPAL = 'FETCH_ROLE_FOR_PRINCIPAL';
export const UPDATE_ROLE = 'UPDATE_ROLE';

export const GET_PRINCIPAL_ACCESS = 'GET_PRINCIPAL_ACCESS';

export const LIST_PERMISSIONS = 'LIST_PERMISSIONS';
export const LIST_APPLICATION_OPTIONS = 'LIST_APPLICATION_OPTIONS';
export const LIST_RESOURCE_OPTIONS = 'LIST_RESOURCE_OPTIONS';
export const LIST_OPERATION_OPTIONS = 'LIST_OPERATION_OPTIONS';

export const GET_RESOURCE_DEFINITIONS = 'GET_RESOURCE_DEFINITIONS';
export const GET_RESOURCE = 'GET_RESOURCE';
