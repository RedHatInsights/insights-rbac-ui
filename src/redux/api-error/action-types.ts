export const API_ERROR = 'API_ERROR' as const;
