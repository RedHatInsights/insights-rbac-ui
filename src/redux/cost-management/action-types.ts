export const FETCH_RESOURCE_DEFINITIONS = 'FETCH_RESOURCE_DEFINITIONS' as const;
export const FETCH_RESOURCE = 'FETCH_RESOURCE' as const;
export const EXPAND_SPLATS = 'EXPAND_SPLATS' as const;
export const RESET_EXPAND_SPLATS = 'RESET_EXPAND_SPLATS' as const;
