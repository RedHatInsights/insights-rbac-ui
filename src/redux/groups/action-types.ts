// Group core actions
export const FETCH_GROUP = 'FETCH_GROUP' as const;
export const FETCH_ADMIN_GROUP = 'FETCH_ADMIN_GROUP' as const;
export const FETCH_SYSTEM_GROUP = 'FETCH_SYSTEM_GROUP' as const;
export const INVALIDATE_SYSTEM_GROUP = 'INVALIDATE_SYSTEM_GROUP' as const;
export const FETCH_GROUPS = 'FETCH_GROUPS' as const;

export const ADD_GROUP = 'ADD_GROUP' as const;
export const UPDATE_GROUP = 'UPDATE_GROUP' as const;
export const REMOVE_GROUPS = 'REMOVE_GROUPS' as const;
export const UPDATE_GROUPS_FILTERS = 'UPDATE_GROUPS_FILTERS' as const;
export const RESET_SELECTED_GROUP = 'RESET_SELECTED_GROUP' as const;

// Group members
export const ADD_MEMBERS_TO_GROUP = 'ADD_MEMBERS_TO_GROUP' as const;
export const REMOVE_MEMBERS_FROM_GROUP = 'REMOVE_MEMBERS_FROM_GROUP' as const;
export const FETCH_MEMBERS_FOR_GROUP = 'FETCH_MEMBERS_FOR_GROUP' as const;
export const FETCH_MEMBERS_FOR_EXPANDED_GROUP = 'FETCH_MEMBERS_FOR_EXPANDED_GROUP' as const;

// Group roles
export const ADD_ROLES_TO_GROUP = 'ADD_ROLES_TO_GROUP' as const;
export const REMOVE_ROLES_FROM_GROUP = 'REMOVE_ROLES_FROM_GROUP' as const;
export const FETCH_ROLES_FOR_GROUP = 'FETCH_ROLES_FOR_GROUP' as const;
export const FETCH_ROLES_FOR_EXPANDED_GROUP = 'FETCH_ROLES_FOR_EXPANDED_GROUP' as const;
export const FETCH_ADD_ROLES_FOR_GROUP = 'FETCH_ADD_ROLES_FOR_GROUP' as const;

// Group service accounts
export const ADD_SERVICE_ACCOUNTS_TO_GROUP = 'ADD_SERVICE_ACCOUNTS_TO_GROUP' as const;
export const REMOVE_SERVICE_ACCOUNTS_FROM_GROUP = 'REMOVE_SERVICE_ACCOUNTS_FROM_GROUP' as const;
export const FETCH_SERVICE_ACCOUNTS_FOR_GROUP = 'FETCH_SERVICE_ACCOUNTS_FOR_GROUP' as const;
