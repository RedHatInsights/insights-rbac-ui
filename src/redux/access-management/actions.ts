import { GET_PRINCIPAL_ACCESS } from './action-types';
import { GetPrincipalAccessParams, getPrincipalAccess as getPrincipalAccessHelper } from './helper';

// Using global ReduxAction interface from store.d.ts
export const getPrincipalAccess = (apiProps: GetPrincipalAccessParams): ReduxAction<Promise<unknown>> => ({
  type: GET_PRINCIPAL_ACCESS,
  payload: getPrincipalAccessHelper(apiProps),
});
