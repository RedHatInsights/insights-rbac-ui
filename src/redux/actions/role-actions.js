import * as ActionTypes from '../action-types';
import * as RoleHelper from '../../helpers/role/role-helper';

export const createRole = (roleData) => ({
  type: ActionTypes.ADD_ROLE,
  payload: RoleHelper.createRole(roleData),
  meta: {
    notifications: {
      fulfilled: {
        variant: 'success',
        title: 'Success adding role',
        dismissDelay: 8000,
        dismissable: false,
        description: 'The role was added successfully.',
      },
      rejected: {
        variant: 'danger',
        title: 'Failed adding role',
        dismissDelay: 8000,
        dismissable: false,
        description: 'The role was not added successfuly.',
      },
    },
  },
});

export const fetchRole = (apiProps) => ({
  type: ActionTypes.FETCH_ROLE,
  payload: RoleHelper.fetchRole(apiProps),
});

export const fetchRoles = (options = {}) => ({
  type: ActionTypes.FETCH_ROLES,
  payload: RoleHelper.fetchRoles(options),
});

export const fetchRolesWithPolicies = (options = {}) => ({
  type: ActionTypes.FETCH_ROLES,
  payload: RoleHelper.fetchRolesWithPolicies(options),
});

export const removeRole = (role) => ({
  type: ActionTypes.REMOVE_ROLE,
  payload: RoleHelper.removeRole(role),
  meta: {
    notifications: {
      fulfilled: {
        variant: 'success',
        title: 'Success removing role',
        dismissDelay: 8000,
        dismissable: false,
        description: 'The role was removed successfully.',
      },
    },
  },
});

export const fetchRoleForUser = (apiProps) => ({
  type: ActionTypes.FETCH_ROLE_FOR_USER,
  payload: RoleHelper.fetchRole(apiProps),
});

export const fetchRoleForPrincipal = (apiProps) => ({
  type: ActionTypes.FETCH_ROLE_FOR_PRINCIPAL,
  payload: RoleHelper.fetchRoleForPrincipal(apiProps),
});

export const fetchRolesForWizard = (options = {}) => ({
  type: ActionTypes.FETCH_ROLES_FOR_WIZARD,
  payload: RoleHelper.fetchRoles(options),
});

export const updateRole = (roleId, data) => ({
  type: ActionTypes.UPDATE_ROLE,
  payload: RoleHelper.updateRole(roleId, data),
});

export const removeRolePermissions = (role, permissionsToRemove) => ({
  type: ActionTypes.UPDATE_ROLE,
  payload: RoleHelper.removeRolePermissions(role, permissionsToRemove),
});
