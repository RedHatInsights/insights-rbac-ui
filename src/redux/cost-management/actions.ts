import { FETCH_RESOURCE, FETCH_RESOURCE_DEFINITIONS } from './action-types';
import { getResource, getResourceDefinitions } from './helper';

// Based on cost API usage patterns, apiProps is typically a configuration object
interface ApiProps {
  [key: string]: unknown;
}

// Using global ReduxAction interface from store.d.ts
export const fetchResourceDefinitions = (apiProps: ApiProps): ReduxAction<Promise<unknown>> => ({
  type: FETCH_RESOURCE_DEFINITIONS,
  payload: getResourceDefinitions(apiProps),
});

export const fetchResource = (apiProps: ApiProps): ReduxAction<Promise<unknown>> => ({
  type: FETCH_RESOURCE,
  payload: getResource(apiProps),
});
