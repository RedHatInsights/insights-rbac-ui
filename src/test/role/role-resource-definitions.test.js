import React from 'react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { fireEvent, render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { FETCH_ROLE } from '../../redux/action-types';
import ResourceDefinitions from '../../smart-components/role/role-resource-definitions';

import * as RoleActions from '../../redux/actions/role-actions';

describe('RoleResourceDefinitions - Cost management', () => {
  let mockStore;
  let initialState;

  const fetchRoleSpy = jest.spyOn(RoleActions, 'fetchRole');

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      roleReducer: {
        isRecordLoading: false,
        selectedRole: {
          uuid: '1234',
          display_name: 'name',
          description: 'description',
          access: [
            {
              resourceDefinitions: Array(21).fill({
                attributeFilter: {
                  key: 'test.test.test',
                  value: 'test',
                  operation: 'equal',
                },
              }),
              permission: 'cost-management:*:read',
            },
          ],
        },
      },
    };
  });

  it('should render resource definitions table', async () => {
    fetchRoleSpy.mockImplementationOnce(() => ({ type: FETCH_ROLE, payload: Promise.resolve({ data: 'something' }) }));
    await act(async () => {
      render(
        <Provider store={mockStore(initialState)}>
          <MemoryRouter initialEntries={['/roles/detail/1234/permission/cost-management:*:read']}>
            <Routes>
              <Route path="/roles/detail/:roleId/permission/:permissionId" element={<ResourceDefinitions />} />
            </Routes>
          </MemoryRouter>
        </Provider>
      );
    });
    expect(screen.getByLabelText('resources table')).toMatchSnapshot();
  });

  it('should not fetch data on filter', async () => {
    fetchRoleSpy.mockImplementationOnce(() => ({ type: FETCH_ROLE, payload: Promise.resolve({ data: 'something' }) }));
    const store = mockStore(initialState);
    await act(async () => {
      render(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/roles/detail/1234/permission/cost-management:*:read']}>
            <Routes>
              <Route path="/roles/detail/:roleId/permission/:permissionId" element={<ResourceDefinitions />} />
            </Routes>
          </MemoryRouter>
        </Provider>
      );
    });
    const expectedActions = [expect.objectContaining({ type: FETCH_ROLE })];
    expect(store.getActions()).toEqual(expectedActions);
    store.clearActions();
    await act(async () => {
      fireEvent.change(screen.getByPlaceholderText('Filter by {key}'), { target: { value: 'c' } });
    });
    expect(store.getActions()).toEqual([]);
  });

  it('should not fetch data on page change', async () => {
    fetchRoleSpy.mockImplementationOnce(() => ({ type: FETCH_ROLE, payload: Promise.resolve({ data: 'something' }) }));
    const store = mockStore(initialState);
    await act(async () => {
      render(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/roles/detail/1234/permission/cost-management:*:read']}>
            <Routes>
              <Route path="/roles/detail/:roleId/permission/:permissionId" element={<ResourceDefinitions />} />
            </Routes>
          </MemoryRouter>
        </Provider>
      );
    });
    const expectedActions = [expect.objectContaining({ type: FETCH_ROLE })];
    expect(store.getActions()).toEqual(expectedActions);
    store.clearActions();
    await act(async () => {
      fireEvent.click(screen.getByLabelText('Go to last page'));
    });
    expect(store.getActions()).toEqual([]);
  });
});

describe('RoleResourceDefinitions - Inventory', () => {
  let mockStore;
  let initialState;

  const fetchRoleSpy = jest.spyOn(RoleActions, 'fetchRole');

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      roleReducer: {
        isRecordLoading: false,
        selectedRole: {
          uuid: '1234',
          display_name: 'name',
          description: 'description',
          access: [
            {
              resourceDefinitions: Array(21).fill({
                attributeFilter: {
                  key: 'group.in',
                  value: [null, 'aaaa', 'bbbb', 'cccc'],
                  operation: 'in',
                },
              }),
              permission: 'inventory:hosts:read',
            },
          ],
        },
      },
    };
  });

  it('should render resource definitions table', async () => {
    fetchRoleSpy.mockImplementationOnce(() => ({ type: FETCH_ROLE, payload: Promise.resolve({ data: 'something' }) }));
    await act(async () => {
      render(
        <Provider store={mockStore(initialState)}>
          <MemoryRouter initialEntries={['/roles/detail/1234/permission/inventory:hosts:read']}>
            <Routes>
              <Route path="/roles/detail/:roleId/permission/:permissionId" element={<ResourceDefinitions />} />
            </Routes>
          </MemoryRouter>
        </Provider>
      );
    });
    expect(screen.getByText('Ungrouped systems')).toBeInTheDocument();
    expect(screen.getByLabelText('resources table')).toMatchSnapshot();
  });
});
