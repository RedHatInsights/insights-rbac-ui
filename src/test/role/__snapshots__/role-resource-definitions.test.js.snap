// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleResourceDefinitions - Cost management should render resource definitions table 1`] = `
<table
  aria-label="resources table"
  class="pf-v5-c-table pf-m-grid-md pf-m-compact"
  data-ouia-component-id="OUIA-Generated-Table-2"
  data-ouia-component-type="PF5/Table"
  data-ouia-safe="true"
  role="grid"
>
  <tbody
    class="pf-v5-c-table__tbody"
    role="rowgroup"
  >
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-1"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-2"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-3"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-4"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-5"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-6"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-7"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-8"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-9"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-10"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-11"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-12"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-13"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-14"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-15"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-16"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-17"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-18"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-19"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-20"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        test
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`RoleResourceDefinitions - Inventory should render resource definitions table 1`] = `
<table
  aria-label="resources table"
  class="pf-v5-c-table pf-m-grid-md pf-m-compact"
  data-ouia-component-id="OUIA-Generated-Table-8"
  data-ouia-component-type="PF5/Table"
  data-ouia-safe="true"
  role="grid"
>
  <tbody
    class="pf-v5-c-table__tbody"
    role="rowgroup"
  >
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-61"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        Ungrouped systems
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-62"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        aaaa
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-63"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        bbbb
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-64"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        cccc
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-65"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        aaaa
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-66"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        bbbb
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-67"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        cccc
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-68"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        aaaa
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-69"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        bbbb
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-70"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        cccc
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-71"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        aaaa
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-72"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        bbbb
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-73"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        cccc
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-74"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        aaaa
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-75"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        bbbb
      </td>
    </tr>
    <tr
      class="pf-v5-c-table__tr"
      data-ouia-component-id="OUIA-Generated-TableRow-76"
      data-ouia-component-type="PF5/TableRow"
      data-ouia-safe="true"
    >
      <td
        class="pf-v5-c-table__td"
        data-key="0"
        data-label="column-0"
        tabindex="-1"
      >
        cccc
      </td>
    </tr>
  </tbody>
</table>
`;
